lua/framework/lua_app.cc

```diff
@@ -53,115 +40,118 @@ LuaApp::InitPetSc(int argc, char** argv)
 int
 LuaApp::Run(int argc, char** argv)
 {
-  InitPetSc(argc, argv);
-  opensn::Initialize();
-  console.PostMPIInfo(opensn::mpi_comm.rank(), opensn::mpi_comm.size());
+  if (opensn::mpi_comm.rank() == 0)
+  {
+    std::cout << opensn::name << " version " << GetVersionStr() << "\n"
+              << Timer::GetLocalDateTimeString() << " Running " << opensn::name << " with "
+              << opensn::mpi_comm.size() << " processes.\n"
+              << opensn::name << " number of arguments supplied: " << argc - 1 << "\n" << std::endl;
+  }
 
-  opensn::log.Log() << opensn::name << " version " << GetVersionStr();
-  opensn::log.Log() << Timer::GetLocalDateTimeString() << " Running " << opensn::name << " with "
-                    << opensn::mpi_comm.size() << " processes.";
-  opensn::log.Log() << opensn::name << " number of arguments supplied: " << argc - 1;
-  opensn::log.LogAll();
-  console.FlushConsole();
+  int error_code = ParseArguments(argc, argv);
 
-  int error_code = 0;
-  ParseArguments(argc, argv);
-  if (not termination_posted_)
+  if (!error_code)
   {
+    InitPetSc(argc, argv);
+    opensn::Initialize();
+    console.PostMPIInfo(opensn::mpi_comm.rank(), opensn::mpi_comm.size());
+    console.FlushConsole();
+
     if (sim_option_interactive_)
       error_code = RunInteractive(argc, argv);
     else
       error_code = RunBatch(argc, argv);
-  }
 
-  opensn::Finalize();
-  PetscFinalize();
+    opensn::Finalize();
+    PetscFinalize();
 
-  opensn::log.Log() << "Elapsed execution time: " << program_timer.GetTimeString();
-  opensn::log.Log() << Timer::GetLocalDateTimeString() << " " << opensn::name
-                    << " finished execution.";
+    if (opensn::mpi_comm.rank() == 0)
+    {
+      std::cout << "\n"
+                << Elapsed execution time: " << program_timer.GetTimeString() << "\n"
+                << Timer::GetLocalDateTimeString() << " " << opensn::name
+                << " finished execution." << std::endl;
+    }
+  }
 
   return error_code;
 }
 
-void
+int
 LuaApp::ParseArguments(int argc, char** argv)
```

Comment:
I think I want to rename (read I want you to rename) this member function to `ProcessArguments`, because that's really what is happening. This function parses the arguments and processes them. Parsing to me means that the arguments are checked syntactically which is what `options.parse(argc, argv);` does.

Reply:
Agreed. Done.